provisioning_templates/provision/coreos_provision.erb

```diff
@@ -1,44 +1,109 @@
 #cloud-config
 <%#
-kind: provision
-name: CoreOS provision
-oses:
-- CoreOS
+kind: snippet
+name: coreos_cloudconfig
 %>
 <%
-os_major  = @host.operatingsystem.major.to_i
-%>
-coreos:
-  units:
-    - name: coreos-bootstrap.service
-      runtime: no
-      command: start
-      content: |
-        [Unit]
-        Description=Install coreos to disk
-        [Service]
-        Type=oneshot
-        ExecStart=/usr/bin/coreos-install \
-          -C <%= @host.operatingsystem.release_name %> \
-          -d <%= @host.params['install-disk'] || '/dev/sda' %> \
-          -c /home/core/cloud-config.yml <% if os_major >= 557 -%>-b <%= @mediapath %><% end %>
-        ExecStartPost=/usr/bin/wget -q -O /dev/null --no-check-certificate <%= foreman_url('built') %>
-        ExecStartPost=/usr/sbin/reboot
-        [X-Fleet]
-        X-Conflicts=coreos-bootstrap.service
-<% if @host.params['ssh_authorized_keys'] -%>
-ssh_authorized_keys:
-<% @host.params['ssh_authorized_keys'].split(',').map(&:strip).each do |ssh_key| -%>
-  - "<%= ssh_key %>"
+  units = 0
+  unless @host.param_false?('enable_etcd')
+    units += 1
+  end
+  if @host.param_true?('expose_docker_socket')
+    units += 1
+  end
+  if @host.subnet.respond_to?(:dhcp_boot_mode?)
+    dhcp = @host.subnet.dhcp_boot_mode? && !@static
+  else
+    dhcp = !@static
+  end
+  unless dhcp
+    units += 1
+  end
+-%>
+<% if units > 0 -%>
+      coreos:
+<% if @host.params['reboot-strategy'] -%>
```

Comment:
Please replace all `@host.params` with `host_param()` as in https://github.com/theforeman/community-templates/pull/484